types/rolling-rate-limiter/index.d.ts

```diff
@@ -0,0 +1,38 @@
+// Type definitions for rolling-rate-limiter 0.1
+// Project: https://github.com/peterkhayes/rolling-rate-limiter
+// Definitions by: Jonas Lochmann <https://github.com/l-jonas>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = RollingRateLimiter;
+
+declare function RollingRateLimiter(options: RollingRateLimiter.InMemoryOptions): RollingRateLimiter.SyncOrAsyncLimiter;
+declare function RollingRateLimiter(options: RollingRateLimiter.WithRedisOptions): RollingRateLimiter.AsyncLimiter;
+
+declare namespace RollingRateLimiter {
+  interface GeneralOptions {
+    interval: number;
+    maxInInterval: number;
+    minDifference?: number;
+  }
+
+  interface RedisOptions {
+    redis: CompatibleRedisClient;
+    namespace?: string;
+  }
+
+  interface CompatibleRedisClient {
+    multi: () => any;
+  }
+
+  type InMemoryOptions = GeneralOptions;
+  type WithRedisOptions = GeneralOptions & RedisOptions;
```

Comment:
Instead of declaring WithRedisOptions as intersection, consider getting rid of RedisOptions and declaring WithRedisOptions  as 
```typescript
interface WithRedisOptions extends GeneralOptions {
    redis: CompatibleRedisClient;
    namespace?: string;
}
```